core/lib/zksync_core/src/consistency_checker/tests/mod.rs

```diff
@@ -298,222 +285,54 @@ async fn normal_checker_function(
 #[tokio::test]
 async fn checker_functions_after_snapshot_recovery(delay_batch_insertion: bool) {
-    let pool = ConnectionPool::test_pool().await;
-    let mut storage = pool.access_storage().await.unwrap();
-    storage
-        .protocol_versions_dal()
-        .save_protocol_version_with_tx(ProtocolVersion::default())
-        .await;
-
-    let l1_batch = create_l1_batch_with_metadata(99);
-
-    let l1_batch_commit_data_generator = Arc::new(RollupModeL1BatchCommitDataGenerator {});
-
-    let commit_tx_input_data = build_commit_tx_input_data(
-        slice::from_ref(&l1_batch),
-        l1_batch_commit_data_generator.clone(),
-    );
-    let client = MockEthereum::default();
-    let signed_tx = client.sign_prepared_tx(
-        commit_tx_input_data.clone(),
-        Options {
-            nonce: Some(0.into()),
-            ..Options::default()
-        },
-    );
-    let signed_tx = signed_tx.unwrap();
-    let commit_tx_hash = signed_tx.hash;
-    client.send_raw_tx(signed_tx.raw_tx).await.unwrap();
-    client.execute_tx(commit_tx_hash, true, 1);
-
-    let save_actions = [
-        SaveAction::InsertBatch(&l1_batch),
-        SaveAction::SaveMetadata(&l1_batch),
-        SaveAction::InsertCommitTx(l1_batch.header.number),
-    ];
-    let commit_tx_hash_by_l1_batch = HashMap::from([(l1_batch.header.number, commit_tx_hash)]);
-
-    if !delay_batch_insertion {
-        for &save_action in &save_actions {
-            save_action
-                .apply(&mut storage, &commit_tx_hash_by_l1_batch)
-                .await;
-        }
-    }
-
-    let (l1_batch_updates_sender, mut l1_batch_updates_receiver) = mpsc::unbounded_channel();
-    let checker = ConsistencyChecker {
-        l1_batch_updater: Box::new(l1_batch_updates_sender),
-        ..create_mock_checker(client, pool.clone())
-    };
-    let (stop_sender, stop_receiver) = watch::channel(false);
-    let checker_task = tokio::spawn(checker.run(stop_receiver, l1_batch_commit_data_generator));
-
-    if delay_batch_insertion {
-        tokio::time::sleep(Duration::from_millis(10)).await;
-        for &save_action in &save_actions {
-            save_action
-                .apply(&mut storage, &commit_tx_hash_by_l1_batch)
-                .await;
-        }
-    }
-
-    // Wait until the batch is checked.
-    let checked_batch = l1_batch_updates_receiver.recv().await.unwrap();
-    assert_eq!(checked_batch, l1_batch.header.number);
-
-    stop_sender.send_replace(true);
-    checker_task.await.unwrap().unwrap();
+    test_helpers::checker_functions_after_snapshot_recovery(
+        delay_batch_insertion,
+        Arc::new(RollupModeL1BatchCommitDataGenerator {}),
+    )
+    .await;
+    test_helpers::checker_functions_after_snapshot_recovery(
+        delay_batch_insertion,
+        Arc::new(ValidiumModeL1BatchCommitDataGenerator {}),
+    )
+    .await;
 }
 
 #[derive(Debug, Clone, Copy)]
-enum IncorrectDataKind {
+pub enum IncorrectDataKind {
```

Comment:
Should this be public for the crate also?

Reply:
I tried, but the test_helpers module didn't recognize the function, so I put `pub` without `(crate)` in this case

Reply:
I have seen the problem, I'm correcting it. Thank you!